### PR TITLE
[Backport] Fix layout xml and page layout caching issue on redis cluster under high load

### PR DESCRIPTION
**Pull request by andrey-legayev**
### Original Pull Request
https://github.com/magento/magento2/pull/22766

### Description (*)
Bugs which were fixed:
 - $this->pageLayout was not checked after reading from cache, but was used as is
 - two cache items were used in once place instead of one (performance impact)

Changes:
 - replace 2 cache items by 1 - it should improve performance
 - add "_MERGED" to cache key suffix to have compatibility with old cache keys during deployment of new version

### Fixed Issues (if relevant)
1. magento/magento2#6942 Layout corruption (Title is wrong, but issue is real)
2. magento/magento2#22976 Corrupt layout cache causes white page on customer account login

### Manual testing scenarios (*)
N/A

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-cicd2** [comment]
[CLA assistant check](https://cla.dev.magento.com/magento/magento2?pullRequest=22625) <br/>All committers have signed the CLA.

**Review by orlangur — APPROVED** [review]
LGTM, previous attempt to fix this issue was performed in https://github.com/magento/magento2/pull/16428#issuecomment-436507767, this one looks better from perfomance perspective.

**Comment by andrey-legayev** [comment]
I've found two issues in previous fix for the issue:
 - It tries to load two cache items even if 1st is not available. It does small impact on performance.
 - Comparison `if ($result && $pageLayoutResult) { ... }` doesn't catch the case when `$pageLayoutResult === ""` (which is ok for Magento layouts) and it generates layouts again and again without using cached data. It had to be `if ($result && $pageLayoutResult !== false) { ... }`

**Comment by andrey-legayev** [comment]
Hi there!
I've submitted changes 2 days ago, but unit tests are still be pending in Travis.
Is it ok?

**Comment by magento-engcom-team** [comment]
@andrey-legayev thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

**Comment by orlangur** [comment]
@andrey-legayev nope, it is not good but EngCom guys are aware of it and it should be fixed soon.

Sorry, just noticed, we need to apply changes against `2.3-develop` first. Could you please prepare such a PR? This one will be `on hold` for now.

**Comment by andrey-legayev** [comment]
@orlangur I've created pull request for 2.3-develop: #22766

**Comment by andrey-legayev** [comment]
This fix has been already included into v2.2.9 by cherry-picking commit.
I'm closing pull request.

**Pull request closed without merging**
